sys/dev/drm/freebsd/drm_os_freebsd.c

```diff
@@ -474,15 +475,20 @@ drm_fstub_do_mmap(struct file *file, const struct file_operations *fops,
 
 		sx_xlock(&drm_vma_lock);
 		TAILQ_FOREACH(dog, &drm_vma_head, link) {
-			if (dog->object->handle == handle)
-				break;
+			if (dog->object->handle == handle) {
+				VM_OBJECT_RLOCK(dog->object);
+				if ((dog->object->flags & OBJ_DEAD) == 0)
```

Comment:
You could perhaps move the `vm_object_reference` up here and then drop the VM object lock sooner?  I'm fine either way though.